cleanlab/internal/util.py

```diff
@@ -688,6 +689,30 @@ def num_unique_classes(labels, multi_label=None) -> int:
         return len(set(labels))
 
 
+def format_labels(labels: LabelLike) -> Tuple[np.ndarray, dict]:
+    """Takes an array of labels and formats it such that labels are in the set ``0, 1, ..., K-1``,
+    where ``K`` is the number of classes. The labels are assigned based on lexicographic order.
+
+    Returns
+    -------
+    labels
+        Returns np.ndarray of shape ``(N,)``. The return labels will be properly formatted and can be passed to other cleanlab functions.
+
+    mapping
+        A dictionary showing the mapping of new to old labels
+    """
+    labels = labels_to_array(labels)
+    if labels.ndim != 1:
+        raise ValueError("labels must be 1D numpy array.")
+
+    unique_labels = np.unique(labels)
+    label_map = {label: i for i, label in enumerate(unique_labels)}
+    formatted_labels = np.array([label_map[l] for l in labels])
+    inverse_map = {i: label for label, i in label_map.items()}
+
+    return formatted_labels, inverse_map
```

Comment:
[nit] no problem since this is helper function, but in future, let's use same variable-name in the returns docstring as in the code, unless there is much clearer but unweidly-long variable-name that we want to use in docstring